src/main/java/com/proiect/pos/controller/MainController.java

```diff
@@ -34,4 +34,33 @@ public ModelAndView showAllProducts(){
         return modelAndView;
     }
 
+
+    @RequestMapping(value = "dashboard", method = RequestMethod.GET)
+    public ModelAndView getAddProductPage() {
+        ModelAndView modelAndView = new ModelAndView();
+        Product product = new Product();
+        modelAndView.addObject("product", product);
+        modelAndView.setViewName("dashboard");
+        return modelAndView;
+    }
+
+    @RequestMapping(value = "dashboard", method = RequestMethod.POST)
```

Comment:
Adaugarea de produs nu se face apeland /dashboard. Te rog schimba pe /admin/createNewProduct.

Dupa care dupa ce se executa functia asta faci un redirect inapoi la Dashboard. Un "NEADMIN" n-ar trebui sa aiba acces sa faca post-uri pe aplicatie.

WebSecurity config se ocupa de asta prin `hasAuthority(Admin)`.
Uite aici, poate te ajuta: https://stackoverflow.com/questions/40880772/spring-boot-redirect-to-a-different-controller-method